homeassistant/components/fritz/__init__.py

```diff
@@ -1 +1,72 @@
-"""The fritz component."""
+"""Support for AVM Fritz!Box functions."""
+import logging
+
+from fritzconnection.core.exceptions import FritzConnectionException, FritzSecurityError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import callback
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .common import FritzBoxTools
+from .const import DOMAIN, SUPPORTED_DOMAINS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up fritzboxtools from config entry."""
+    _LOGGER.debug("Setting up FRITZ!Box Tools component")
+    fritz_tools = FritzBoxTools(
+        hass=hass,
+        host=entry.data[CONF_HOST],
+        port=entry.data[CONF_PORT],
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+    )
+
+    try:
+        await fritz_tools.async_setup()
+        await fritz_tools.async_start()
+    except FritzSecurityError as ex:
+        raise ConfigEntryAuthFailed from ex
+    except FritzConnectionException as ex:
+        raise ConfigEntryNotReady from ex
+
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = fritz_tools
+
+    @callback
+    def _async_unload(event):
+        fritz_tools.async_unload()
+
+    entry.async_on_unload(
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _async_unload)
+    )
+    # Load the other platforms like switch
+    for domain in SUPPORTED_DOMAINS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, domain)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistantType, entry: ConfigType) -> bool:
+    """Unload FRITZ!Box Tools config entry."""
+    fritzbox: FritzBoxTools = hass.data[DOMAIN][entry.entry_id]
+    fritzbox.async_unload()
+
```

Comment:
We can unload them in one go instead of sequentially. 

```suggestion
    unload_ok = all(
        await asyncio.gather(
            *[
                hass.config_entries.async_forward_entry_unload(entry, platform)
                for platform in SUPPORTED_DOMAINS
            ]
        )
    )

    if unload_ok:
        hass.data[DOMAIN].pop(entry.entry_id)
        
    return unload_ok
```

Also rename `SUPPORTED_DOMAINS` to `PLATFORMS`

Reply:
Optimized